website/static/js/accountSettings.js

```diff
@@ -205,7 +211,8 @@ var UserProfileViewModel = oop.extend(ChangeMessageMixin, {
                         return;
                     }
                 }
-                this.changeMessage('Invalid Email.', 'text-danger');
+            }.bind(this)).fail(function(){
+                this.profile().emails.remove(email);
```

Comment:
That said, updating the UI optimistically--i.e. adding the email before the API request has finished--is good for usability, so I think this solution might be ok.